src/Microsoft.Identity.Web.AgentIdentities/AgentIdentitiesExtension.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Identity.Abstractions;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Extensions methods to enable agent identities in Microsoft.Identity.Web.
+    /// </summary>
+    public static class AgentIdentityExtension
+    {
+        /// <summary>
+        /// Enable support for agent identities.
+        /// </summary>
+        /// <param name="services">Service collection</param>
+        /// <returns>The service collection for chaining.</returns>
+        public static IServiceCollection AddAgentIdentities(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services), "Service collection cannot be null.");
+            }
+
+            // Register the OidcFic services for agent applications to work.
+            services.AddOidcFic();
+
+            return services;
+        }
+
+
+        /// <summary>
+        /// Updates the options to acquire a token for the agent identity.
+        /// </summary>
+        /// <param name="options">Authorization header provider options.</param>
+        /// <param name="agentApplicationId">The agent identity GUID.</param>
+        /// <returns>The updated authorization header provider options.</returns>
+        public static AuthorizationHeaderProviderOptions WithAgentIdentity(this AuthorizationHeaderProviderOptions options, string agentApplicationId)
+        {
+            // It's possible to start with no options, so we initialize it if it's null.
+            if (options == null)
+                options = new AuthorizationHeaderProviderOptions();
+
+            // AcquireTokenOptions holds the information needed to acquire a token for the Agent Identity
+            if (options.AcquireTokenOptions == null)
+                options.AcquireTokenOptions = new AcquireTokenOptions();
+
+            options.AcquireTokenOptions.ForAgentIdentity(agentApplicationId);
+
+            return options;
+        }
+
+        // TODO:make public?
```

Comment:
pls remove TODOs.

Reply:
that's another scenario I want to revisit (with Token Acquirer only)